docs/DeveloperGuide.md

```diff
@@ -1,35 +1,306 @@
 # Developer Guide
```

Comment:
Overall, I like the sequence and architecture diagrams. Perhaps can consider adding in class diagrams as well